### PR TITLE
Improved a11y for Layer, Select, TextArea and TextInput

### PR DESCRIPTION
**Pull request by alansouzati**
#### What does this PR do?
Changes Layer Select TextArea and TextInput to have a better accessibility.

Today pressing esc inside a Select closes the Layer. So I decided to take a pass in this feature for all components.

#### Where should the reviewer start?
LayerContainer

#### What testing has been done on this PR?
manual

#### How should this be manually tested?
open layer stories and select stories and make sure you can use the keyboard consistently

#### What are the relevant issues?
no issue, sorry

#### Do the grommet docs need to be updated?
no

#### Should this PR be mentioned in the release notes?
yes

#### Is this change backwards compatible or is it a breaking change?
backwards

### CODE REVIEW

File: src/js/components/Drop/DropContainer.js [hunk 1]
```diff
@@ -292,7 +301,7 @@ export class DropContainer extends Component {
 
     return (
       <FocusedContainer>
-        <Keyboard onEsc={onEsc} onKeyDown={onKeyDown} target="document">
```
Comment (ericsoderberghp) [inline src/js/components/Drop/DropContainer.js]: It doesn't seem right to remove the document target here. Doesn't that prevent Esc from closing the drop when the drop might not have focus? For example, if Drop has `onClickOutside={false}` but wants Esc to close the drop.

File: src/js/components/Layer/LayerContainer.js [hunk 2]
```diff
@@ -51,6 +60,9 @@ class LayerContainer extends Component {
     const { position } = this.props;
     if (position !== 'hidden') {
       this.makeLayerVisible();
+      if (this.anchorRef.current) {
+        this.anchorRef.current.focus();
```
Comment (ericsoderberghp) [inline src/js/components/Layer/LayerContainer.js]: Does this prevent the caller from also trying to set focus on a particular input element in the Layer?

File: src/js/components/TextArea/TextArea.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ import { StyledTextArea } from './StyledTextArea';
 
 class TextArea extends Component {
   onEsc = event => {
+    event.stopPropagation();
```
Comment (ericsoderberghp) [inline src/js/components/TextArea/TextArea.js]: Why was this needed?

File: src/js/components/Drop/DropContainer.js (hunk 1 above)
Comment (alansouzati) [reply src/js/components/Drop/DropContainer.js]: correct let me take a look

File: src/js/components/Layer/LayerContainer.js (hunk 2 above)
Comment (alansouzati) [reply src/js/components/Layer/LayerContainer.js]: I dont think so since componentDidMount on layercontainer will always be called first.

File: src/js/components/TextArea/TextArea.js (hunk 3 above)
Comment (alansouzati) [reply src/js/components/TextArea/TextArea.js]: take a look here
https://storybook.grommet.io/?selectedKind=Layer&selectedStory=Form&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel

go to the input and try pressing esc inside the layer. the layer will close.

I don't think we should be using stopImmediatePropagation.

Comment (ericsoderberghp) [reply src/js/components/TextArea/TextArea.js]: The Layer isn't closing for me in the Layer Form story.

File: src/js/components/Layer/LayerContainer.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Layer/LayerContainer.js]: Can you at least add a comment explaining why we add this anchor and give it focus?

File: src/js/components/Drop/DropContainer.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/Drop/DropContainer.js]: I'm a bit lost at what we're doing here. It seems rather convoluted. Can you include some comments explaining it?

File: src/js/components/Layer/LayerContainer.js [hunk 4]
```diff
@@ -121,14 +143,11 @@ class LayerContainer extends Component {
           {content}
         </StyledLayer>
       );
+      /* eslint-enable jsx-a11y/anchor-is-valid, jsx-a11y/anchor-has-content */
```
Comment (ericsoderberghp) [inline src/js/components/Layer/LayerContainer.js]: Remove?

File: src/js/components/Layer/LayerContainer.js [hunk 5]
```diff
@@ -9,6 +9,13 @@ import { backgroundIsDark } from '../../utils';
 
 import { StyledLayer, StyledContainer, StyledOverlay } from './StyledLayer';
 
+const hiddenAnchor = {
```
Comment (ericsoderberghp) [inline src/js/components/Layer/LayerContainer.js]: Can we do this via styled-components rather than inline styling?
